ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -1737,6 +1735,18 @@ public int execute(boolean deferClose) throws CommandNeedRetryException {
 
     try {
       LOG.info("Executing command(queryId=" + queryId + "): " + queryStr);
+
+      int webuiPort = conf.getIntVar(ConfVars.HIVE_SERVER2_WEBUI_PORT);
```

Comment:
I was reading the code a little more, why dont we put this in SqlOperation.prepare (which would run at the point you do right now), or even before SqlOperation.runQuery?

That way we dont have to modify the driver.

SQlOperation will be all statements visible to webui (the other operations are if you use a JDBC and happen to call getColumns on HS2, but this doesnt happen at all)